components/CreateSetupLink.tsx

```diff
@@ -0,0 +1,244 @@
+import { FormEvent, useState } from 'react';
+import Link from 'next/link';
+import {
+  CheckCircleIcon,
+  ExclamationCircleIcon,
+  ClipboardIcon,
+  ArrowLeftIcon,
+} from '@heroicons/react/24/outline';
+import ConfirmationModal from '@components/ConfirmationModal';
+
+const CreateSetupLink = (props: { service: 'Jackson' | 'Directory-Sync' }) => {
+  const createSetupLink = async (event) => {
```

Comment:
Method name `createSetupLink` can be changed to `createLink` so that it doesn't cause confusion with the component name and aligns with `regenerateLink`

Reply:
How about renaming regenerateLink to regenerateSetupLink?

Reply:
For me the component name already captures the "type" of link being created. Having the same name for the event handler/method might trip someone who is skimming through the code.